test/Acceptance_Tests/answer_endpoints.js

```diff
@@ -9,10 +9,68 @@ var app = require('../../server');
 var jwt = require('jwt-simple');
 var User = require('../../models/user');
 var Question = require('../../models/question');
-var dbUtils = require('../test_db_utils');
+var Answer = require('../../models/answer');
+var dbUtils = require('../../utils/databaseutils');
+var Constants = require('../../constants');
 
 describe('Answer Endpoints', function () {
-    describe('POST /answers?', function () {
-        it("should not post if missing parameters");
+    describe("POST /answers", function () {
+        var question;
+        var user;
+        before(function (done) {
+            Answer.remove({}, function(err) {
+                dbUtils.createTestUser("email", "password", "firstName", function (newUser) {
+                    user = newUser;
+                    dbUtils.createQuestion("title", "description", user._id, 1, function (doc) {
+                        question = doc;
+                        done();
+                    });
+                });
+            });
+        });
+        after(function (done) {
+            dbUtils.clearUserCollection(function () {
+                dbUtils.clearQuestionCollection(function() {
+                    Answer.remove({}, done);
+                });
+            });
+        });
+        it("should not post if missing questionId", function (done) {
+            request(app).post('/api/answers').expect(400).end(function (err, res) {
+                res.body.error.should.equal(Constants.ERROR.MISSING.QUESTION_ID);
+                done();
+            });
+        });
+        it("should not post if missing answerBody", function (done) {
+            request(app).post('/api/answers').send({questionId: "someId"}).expect(400).end(function (err, res) {
+                res.body.error.should.equal(Constants.ERROR.MISSING.ANSWER_BODY);
+                done();
+            });
+        });
+        it("should not post if missing token", function (done) {
+            request(app).post('/api/answers').send({questionId: "someId", answerBody: "description"}).expect(400).end(function (err, res) {
+                res.body.error.should.equal(Constants.ERROR.MISSING.TOKEN);
+                done();
+            });
+        });
+        it("should not post if invalid token", function (done) {
+            request(app).post('/api/answers').send({questionId: "someId", answerBody: "description", token: "blahblahblah"}).expect(401).end(function (err, res) {
+                res.body.error.should.equal(Constants.ERROR.INVALID.TOKEN);
+                done();
+            });
+        });
+        it("should post answer when given valid info", function (done) {
```

Comment:
Shouldn't the when be an if? Doesn't really matter but lets pick one and stick to it.